pySDC/implementations/problem_classes/generic_spectral.py

```diff
@@ -0,0 +1,357 @@
+from pySDC.core.problem import Problem, WorkCounter
+from pySDC.helpers.spectral_helper import SpectralHelper
+import numpy as np
+from pySDC.core.errors import ParameterError
+
+
+class GenericSpectralLinear(Problem):
+    """
+    Generic class to solve problems of the form M u_t + L u = y, with mass matrix M, linear operator L and some right hand side y using spectral methods.
+    L may contain algebraic conditions, as long as (M + dt L) is invertible.
+
+    """
+
+    @classmethod
+    def setup_GPU(cls):
+        """switch to GPU modules"""
+        import cupy as cp
+        from pySDC.implementations.datatype_classes.cupy_mesh import cupy_mesh, imex_cupy_mesh
+        from pySDC.implementations.datatype_classes.mesh import mesh, imex_mesh
+
+        cls.dtype_u = cupy_mesh
+
+        GPU_versions = {
+            mesh: cupy_mesh,
+            imex_mesh: imex_cupy_mesh,
+        }
+
+        cls.dtype_f = GPU_versions[cls.dtype_f]
+
+    def __init__(
```

Comment:
Parameter documentation should be put with the class documentation, not in the `__init__` docstring. Also, the `setup_GPU` should be mentioned in the class documentation (with a short description of its usage)

Reply:
And furthermore, `*args` and `**kwargs` are not used in the constructor, please remove them (just open the possibility to pass a wrong named argument that won't be taken into account ...)